### PR TITLE
[wasm] Add ArgMin

### PR DESCRIPTION
**Pull request by chunnienc**
To see the logs from the Cloud Build CI, please join either our [discussion](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs) or [announcement](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs-announce) mailing list.

- - -
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/7306)

### CODE REVIEW

File: tfjs-backend-wasm/src/kernels/ArgMin.ts [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+  const innerSize = input.shape[axes[0]];
+  wasmFunc(inputId, CppDType[input.dtype], outerSize, innerSize, outId);
+
+  if (inputWasTransposed) {
+    // dispose of the transposed tensor.
+    backend.disposeData(transposed.dataId);
+  }
+
+  return out;
+}
```
Comment (mattsoulanille) [inline tfjs-backend-wasm/src/kernels/ArgMin.ts:74]: Nit: This looks similar to ArgMax's implementation. Can we factor it out? If other ops have patterns like this too, then maybe this is better to do in a different PR.

File: tfjs-backend-wasm/src/cc/kernels/ArgMin.cc [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+namespace {
+
+template <typename T>
+void ArgMinImpl(const T* x, const int outer_size, const int inner_size,
+                int32_t* out_buf) {
+  for (int i = 0; i < outer_size; ++i) {
+    const int offset = i * inner_size;
+    T min_value = x[offset];
+    int min_index = 0;
```
Comment (mattsoulanille) [inline tfjs-backend-wasm/src/cc/kernels/ArgMin.cc:34]: Maybe we can factor out some of this array indexing code. It seems like several ops act on just the last dimension of the tensor. Could we have a function that applies another function to the last dimension (or maybe a more general one that works on the last n dimensions)?

Then, we could just pass it this function.

```c
void ArgMaxInner(const T* x, const int size, int32_t* out_buf ) {
  T min_value = x[0];
  int min_index = 0;
  for (int i = 1; i < size; i++) {
    t Value = x[i];
    if (value < min_value) {
      min_value = value;
      min_index = i;
  }
  out_buf[i] = min_index;
}
```

Probably better for a later PR, though.

**Review by mattsoulanille — APPROVED** [review]
LGTM. I have some ideas about how we might simplify some of these kernels that could be applied to future PRs, but this PR looks good to me.

**Review by pyu10055 — APPROVED** [review]
> Reviewed 7 of 7 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/7306)* status: :shipit: complete! 2 of 1 approvals obtained (waiting on @chunnienc)

**Comment by mattsoulanille** [comment]
@chunnienc Feel free to merge this whenever you like.

**Comment by chunnienc** [comment]
> @chunnienc Feel free to merge this whenever you like.

I'm working on the common utils you mentioned. Will append to this PR and request your review again once ready.

File: tfjs-backend-wasm/src/cc/kernels/ArgMin.cc (hunk 2 above)
Comment (chunnienc) [reply tfjs-backend-wasm/src/cc/kernels/ArgMin.cc:34]: Added

**Review by mattsoulanille — APPROVED** [review]
LGTM
